### PR TITLE
Expose prometheus metrics endpoint to report e2e latency and gas usage

### PR DESCRIPTION
**Pull request by Inkvi**
Example of reported metrics at `/metrics`:
```
# HELP txLatency Optimism E2E Tx Latency in seconds
# TYPE txLatency gauge
txLatency{chainName="optimism",statName="Avg"} 26
txLatency{chainName="optimism",statName="Min"} 26
txLatency{chainName="optimism",statName="Median"} 26
txLatency{chainName="optimism",statName="Max"} 26

# HELP ackTransactionCost Optimism Ack Gas Cost in gwei
# TYPE ackTransactionCost gauge
ackTransactionCost{chainName="optimism",statName="Avg"} 219.222230084
ackTransactionCost{chainName="optimism",statName="Min"} 219.222230084
ackTransactionCost{chainName="optimism",statName="Median"} 219.222230084
ackTransactionCost{chainName="optimism",statName="Max"} 219.222230084

# HELP sendPacketTransactionCost Optimism Send Packet Gas Cost in gwei
# TYPE sendPacketTransactionCost gauge
sendPacketTransactionCost{chainName="optimism",statName="Avg"} 91.937162328
sendPacketTransactionCost{chainName="optimism",statName="Min"} 91.937162328
sendPacketTransactionCost{chainName="optimism",statName="Median"} 91.937162328
sendPacketTransactionCost{chainName="optimism",statName="Max"} 91.937162328
```

## Summary by CodeRabbit

- **New Features**
	- Introduced a new metrics route for Prometheus metrics.
	- Updated `CHAIN_CONFIGS` with new RPC URLs for "Optimism" and "Base" chains.
	- Added utility functions for blockchain data analysis and metrics calculation.

- **Bug Fixes**
	- Improved packet sequencing in the `getPackets` function by adding timestamp checks.
	- Added a condition to ensure `ackLog`'s `blockNumber` is greater than `sendPacketLog`'s before processing.

- **Refactor**
	- Removed unnecessary logging in `CachingJsonRpcProvider`.
	- Reorganized and refactored code related to metrics calculation and data fetching in `app/page.tsx`.

### CODE REVIEW

File: app/chains.ts [hunk 1]
```diff
@@ -12,15 +12,15 @@ export const CHAIN_CONFIGS: {
   optimism: {
     id: 11155420,
     display: "Optimism",
-    rpc: "https://opt-sepolia.g.alchemy.com/v2/RN7slh_2cUIuzhxo4M9VgYCbqRcPOmkJ",
+    rpc: "https://opt-sepolia.g.alchemy.com/v2/jKvLhhXvtnWdNeZrKst0demxnwJcYH1o",
```
Comment (Inkvi) [inline app/chains.ts:15]: Update to paid RPCs

File: app/api/ibc/[type]/packets.ts [hunk 2]
```diff
@@ -194,11 +198,18 @@ export async function getPackets(request: NextRequest, apiUrl: string) {
 
 
     const key = `0x${channel.portId.split(".")[2]}-${channel.channelId}-${sequence}`;
+    const recvBlock = await providerTo.getBlock(recvPacketLog.blockNumber)
+
+    if (recvBlock!.timestamp < packets[key].createTime) {
+      continue
+    }
+
```
Comment (Inkvi) [inline app/api/ibc/[type]/packets.ts:206]: Bug fix for when a recv packet event has a timestamp earlier than send packet event

File: app/api/metrics/route.ts [hunk 3]
```diff
@@ -47,7 +47,9 @@ const createLogPairs = (ackLogs: Array<ethers.EventLog>, sendPacketLogs: Array<e
 
     if (ackLogsMap.has(key)) {
       const ackLog = ackLogsMap.get(key)!;
-      logPairs.push({ackLog, sendPacketLog});
+      if (ackLog.blockNumber > sendPacketLog.blockNumber) {
+        logPairs.push({ackLog, sendPacketLog});
+      }
```
Comment (Inkvi) [inline app/api/metrics/route.ts:52]: Bug fix for when a ack packet event has a timestamp earlier than send packet event

**Comment by dshiell** [comment]
@coderabbitai review

File: app/metrics/route.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        gauges[metric.key] = gauge
+      }
+    }
+  }
+
+  let defaultMetrics = await register.metrics();
+
+  return new Response(defaultMetrics, {
+    headers: {
+      'Cache-Control': 'no-store',
```
Comment (dshiell) [inline app/metrics/route.ts:43]: do we not want it cached? seems like it might be worth caching for a bit?

**Comment by dshiell** [comment]
cc: @mvpoyatt

File: app/metrics/route.ts (hunk 4 above)
Comment (Inkvi) [reply app/metrics/route.ts:43]: The caching of metrics seems wrong to me. It's something that always changing and the collector should always want to get the most up-to-date data points.

FYI we cache the responses from the rpc provider about the block information.

**Review by mvpoyatt — APPROVED** [review]

**Review by mvpoyatt — CHANGES_REQUESTED** [review]
Updating review status re comment about caching

**Review by mvpoyatt — APPROVED** [review]

**Pull request merged**